### ISSUE TITLE
custom sequential schedulers

### ISSUE THREAD

**Issue by naomi-simumba** [IBM/terratorch#655]
**Is your feature request related to a problem? Please describe.**
current implementation of `optimizer_factory`  only allows for optimisers from the `torch.optim.lr_scheduler` class. 
https://github.com/IBM/terratorch/blob/25fc5e6969d680edd8e5e779f98d6a2744b69c5d/terratorch/tasks/optimizer_factory.py#L9

More complex scheduling may be achieved with the SequentialLR scheduler, which uses multiple schedulers. However, this is cannot be used with the current set-up.

**Describe the solution you'd like**
Add functionality to unpack Sequential schedulers. e.g. <link> 

**Describe alternatives you've considered (optional)**
Users create custom tasks. This would mean copying the entire existing task and changing only the `configure_optimizers` fn.

**Additional context (optional)**
Several newer schedulers are being used in the community and are not available from `torch`. However, these may be available from other libraries. e.g. `timm`.
